backend-elasticsearch/src/main/java/org/hibernate/search/v6poc/backend/elasticsearch/document/model/impl/AbstractElasticsearchTypedFieldMetadataContext.java

```diff
@@ -8,14 +8,14 @@
 
 import org.hibernate.search.v6poc.backend.document.impl.DeferredInitializationIndexFieldAccessor;
 import org.hibernate.search.v6poc.backend.document.spi.IndexFieldAccessor;
-import org.hibernate.search.v6poc.backend.elasticsearch.document.model.ElasticsearchTypedFieldModelContext;
+import org.hibernate.search.v6poc.backend.elasticsearch.document.model.ElasticsearchTypedFieldMetadataContext;
 import org.hibernate.search.v6poc.backend.elasticsearch.document.model.impl.esnative.PropertyMapping;
 
 /**
  * @author Yoann Rodiere
  */
-public abstract class AbstractElasticsearchTypedFieldModelContext<T>
-		implements ElasticsearchTypedFieldModelContext<T>, ElasticsearchIndexModelNodeContributor<PropertyMapping> {
+public abstract class AbstractElasticsearchTypedFieldMetadataContext<T>
+		implements ElasticsearchTypedFieldMetadataContext<T>, ElasticsearchIndexModelNodeContributor<PropertyMapping> {
```

Comment:
But you still have `ElasticsearchIndexModelNodeContributor`?

I thought that maybe you were using `Model` for the whole thing and `Metadata` for the separate elements but it doesn't seem to be the case. I think it wouldn't be a bad thing to do so tbh.

Reply:
Do you have a concrete suggestion? Because it's not easy to find a correct name TBH...

 * `ElasticsearchIndexMetadataContributor` is really ambiguous, because (at least currently) the type doesn't allow to alter the index metadata (which are a different thing in Elasticsearch). Only the "mapping" (in ES wording) / "schema" / "index model". Also
 * `ElasticsearchIndexFieldMetadataContributor` is not correct, because this type is also used to contribute metadata that do not apply to any field in particular. Whether the "_routing" attribute will be mandatory when adding a document to this index, for instance.

With that in mind, "IndexModelNode" really translates what I meant: the implementations contribute to a *part* of the *index model*.

I guess we could go with `ElasticsearchTypeMappingNodeContributor` or something like that if you prefer, relying on the Elasticsearch wording to make the distinction more obvious?